src/main/java/com/redislabs/redisgraph/impl/Utils.java

```diff
@@ -68,6 +65,38 @@ public static String prepareQuery(String query, Object ...args){
         return query;
     }
 
+    /**
+     * Prepare and formats a query and query arguments
+     * @param query - query
+     * @param params - query parameters
+     * @return query with parameters header
+     */
+    public static String prepareQuery(String query, Map<String, Object> params){
+        StringBuilder sb = new StringBuilder("CYPHER ");
+        for(Map.Entry<String, Object> entry : params.entrySet()) {
+            String key = entry.getKey();
+            Object value = entry.getValue();
+            sb.append(key).append("=");
+            sb.append(valueToString(value));
+            sb.append(" ");
+        }
+        sb.append(query);
+        return sb.toString();
+    }
+
+    private static String valueToString(Object value) {
+        if(value == null)
+            return "null";
+        Class valueClass = value.getClass();
+        if(valueClass == String.class){
+            return quoteString((String) value);
+        }
+        if(valueClass.isArray()){
+            return Arrays.toString((Object[]) value);
```

Comment:
Do we need to use quoteString on each value in the array?

Reply:
good catch
changed